vcd/resource_vcd_nsxt_security_group_test.go

```diff
@@ -0,0 +1,477 @@
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  name = "test-security-group-changed"
+}
+`
+
+const testAccNsxtSecurityGroupDatasource = testAccNsxtSecurityGroup + `
+# skip-binary-test: Terraform resource cannot have resource and datasource in the same file
+data "vcd_nsxt_security_group" "group1" {
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+
+	edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+	name            = "test-security-group"
+}
+`
+
+// TestAccVcdNsxtSecurityGroupInvalidConfigs is expected to fail when:
+// * NSX-V Edge Gateway ID is supplied
+// * Invalid (non existent) Edge Gateway ID is presented
+// * Isolated Org Vdc network added as a member
+func TestAccVcdNsxtSecurityGroupInvalidConfigs(t *testing.T) {
+	preTestChecks(t)
+	skipNoNsxtConfiguration(t)
+
+	// This test is meant to fail
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	// String map to fill the template
+	var params = StringMap{
+		"Org":         testConfig.VCD.Org,
+		"NsxtVdc":     testConfig.Nsxt.Vdc,
+		"EdgeGw":      testConfig.Networking.EdgeGateway,
+		"NetworkName": t.Name(),
+		"Tags":        "network nsxt",
+	}
+
+	configText := templateFill(testAccVcdNsxtSecurityGroupIncorrectEdgeGateway, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 1: %s", configText)
+
+	params["FuncName"] = t.Name() + "-step2"
+	configText1 := templateFill(testAccVcdNsxtSecurityGroupIncorrectEdgeGatewayStep2, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 2: %s", configText1)
+
+	params["FuncName"] = t.Name() + "-step3"
+	configText2 := templateFill(testAccVcdNsxtSecurityGroupIncorrectEdgeGatewayStep3, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 3: %s", configText2)
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		PreCheck:          func() { testAccPreCheck(t) },
+		CheckDestroy: resource.ComposeAggregateTestCheckFunc(
+			testAccCheckNsxtFirewallGroupDestroy(testConfig.Nsxt.Vdc, "test-security-group", types.FirewallGroupTypeSecurityGroup),
+		),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:      configText,
+				ExpectError: regexp.MustCompile(`please use 'vcd_nsxt_edgegateway' for NSX-T backed VDC`),
+			},
+			resource.TestStep{
+				Config:      configText1,
+				ExpectError: regexp.MustCompile(`error creating NSX-T Security Group`),
+			},
+			resource.TestStep{
+				Config:      configText2,
+				ExpectError: regexp.MustCompile(`Not all member network IDs reference to Routed Org networks`),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdNsxtSecurityGroupIncorrectEdgeGateway = `
+data "vcd_edgegateway" "existing" {
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+
+	name = "{{.EdgeGw}}"
+}
+
+resource "vcd_nsxt_security_group" "group1" {
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+  
+	edge_gateway_id = data.vcd_edgegateway.existing.id
+  
+	name = "test-security-group"
```

Comment:
This script is nor well formatted

Reply:
Reformatted all scripts with `terraform fmt` in this file.